src/transaction.js

```diff
@@ -52,174 +66,178 @@ function convertAmountToMinimumUnit(amount_str) {
-    //     throw new Error("This contract is not supported in the current SDK version");
-    // }
     let init_data = cold_tx['initData'];
-    let amount = init_data['amount'];
-    let unity = init_data['unity'];
-    let token_description = init_data['token_description'];
     let public_key_bytes = Base58.decode(cold_tx['senderPublicKey']);
     cold_tx['address'] = acc.convertPublicKeyToAddress(public_key_bytes, network_byte);
-    if(cold_tx['contract'] === Contract.TOKEN_CONTRACT || cold_tx['contract'] === Contract.TOKEN_CONTRACT_WITH_SPLIT) {
-        if(!init_data.hasOwnProperty('amount')) {
-            throw new Error("There is no field 'amount' in initData");
-        }
-        if(!init_data.hasOwnProperty('unity')) {
-            throw new Error("There is no field 'unity' in initData");
-        }
-        if(!init_data.hasOwnProperty('token_description')) {
-            throw new Error("There is no field 'token_description' in initData");
-        }
-        cold_tx['contractInitExplain'] = 'Create token' + (cold_tx['contract'] === Contract.TOKEN_CONTRACT ? ' ' : ' (support split) ') + 'with max supply ' + BigNumber(amount);
-        cold_tx['contractInitTextual'] = "init(max=" + BigNumber(amount) + ",unity= "+ BigNumber(unity) + ",tokenDescription='" + token_description + "')";
-        cold_tx['contractInit'] = processContractData(init_data);
-    } else {
-        cold_tx['contractInitExplain'] = ''
-        cold_tx['contractInitTextual'] = ''
-        cold_tx['contractInit'] = init_data
+    let contract_type = getContractType(cold_tx['contract']);
+    switch (contract_type) {
+        case 'TOKEN_CONTRACT': case 'TOKEN_CONTRACT_WITH_SPLIT':
+            cold_tx['contractInitExplain'] = 'Create token' + (contract_type === 'TOKEN_CONTRACT' ? ' ' : ' (support split) ') + 'with max supply ' + BigNumber(init_data[0]['value']).dividedBy(init_data[1]['value']);
+            cold_tx['contractInitTextual'] = "init(max=" + BigNumber(init_data[0]['value']).dividedBy(init_data[1]['value'])+ ",unity= "+ BigNumber(init_data[1]['value']) + ",tokenDescription='" + init_data[2]['value'] + "')";
+            break;
+        case 'PAYMENT_CONTRACT': case 'LOCK_CONTRACT':
+            cold_tx['contractInitExplain'] = ''
+            cold_tx['contractInitTextual'] = ''
     }
+    cold_tx['contractInit'] = processData(init_data);
     delete cold_tx['senderPublicKey'];
     delete cold_tx['initData'];
 }
 
-function getFunctionColdFields(cold_tx, stored_tx, function_type, network_byte, acc) {
+function getFunctionColdFields(cold_tx, network_byte, acc) {
     let init_data = cold_tx['functionData'];
-    switch (function_type) {
+    let function_index = cold_tx['functionIndex'];
+    let num, data;
+    switch (function_index) {
         case Constants.SUPERSEDE_FUNCIDX:
-            cold_tx['functionExplain'] = 'Set issuer to ' + init_data['new_issuer'];
+            cold_tx['functionExplain'] = 'Set issuer to ' + init_data[0]['value'];
```

Comment:
Remove `functionExplain` in SDK. The explain is open field which implement in application (For example, web hot wallet).